src/pipeline.js

```diff
@@ -112,9 +112,10 @@ class Pipeline {
      * Registers an extension to the pipeline.
      * @param {String} name - name of the extension point (typically the function name).
      * @param {pipelineFunction} f - a new pipeline step that will be injected relative to `name`.
-     * @param {integer} offset - where to insert the new function (0 = before, 1 = after)
+     * @param {integer} before - where to insert the new function (true = before, false = after)
      */
-    this.attach.generic = (name, f, offset = 0) => {
+    this.attach.generic = (name, f, before = true) => {
```

Comment:
pedantic comment: good practice is to have `false` as default...but since the invention of defaults for arguments, this is no longer so important :-)

Reply:
btw: do you really want to export `attach.generic` ?

Reply:
No.